src/components/app/details/triggerView/cdMaterial.tsx

```diff
@@ -1,231 +1,852 @@
+                                <span className="fs-13 fw-4">{mat.deployedBy}</span>
+                            </>
+                        )}
+                    </div>
+                ) : (
+                    <div />
+                )}
+                {!hideSelector && (
+                    <div className="material-history__select-text w-auto">
+                        {mat.vulnerable ? (
+                            <span className="material-history__scan-error">Security Issues Found</span>
+                        ) : mat.isSelected ? (
+                            <Check className="dc__align-right icon-dim-24" />
+                        ) : (
+                            'Select'
+                        )}
+                    </div>
+                )}
+            </>
+        )
+    }
+
+    renderMaterial() {
+        const materialList =
+            this.state.isRollbackTrigger && this.state.showOlderImages ? [this.props.material[0]] : this.props.material
+        return materialList.map((mat, index) => {
+            return (
+                <div
+                    key={`material-history-${index}`}
+                    className={`material-history material-history--cd ${
+                        mat.isSelected ? 'material-history-selected' : ''
+                    }`}
+                >
+                    {this.renderSequentialCDCardTitle(mat)}
+                    <div
+                        className={`material-history__top ${
+                            !this.state.isSecurityModuleInstalled && mat.showSourceInfo ? 'dc__border-bottom' : ''
+                        }`}
+                        style={{ cursor: `${mat.vulnerable ? 'not-allowed' : mat.isSelected ? 'default' : 'pointer'}` }}
+                        onClick={(event) => {
+                            event.stopPropagation()
+                            if (!mat.vulnerable) {
+                                this.handleImageSelection(index, mat)
+                            }
+                        }}
+                    >
+                        {this.renderMaterialInfo(mat)}
+                    </div>
+                    {mat.showSourceInfo && (
                         <>
-                            <span className="cn-9 fw-4" style={{ fontStyle: 'italic' }}>
-                                {' '}
-                                and{' '}
-                            </span>
-                            {this.props.envName}
+                            {this.state.isSecurityModuleInstalled && !this.props.hideInfoTabsContainer && (
+                                <ul className="tab-list tab-list--vulnerability">
+                                    <li className="tab-list__tab">
+                                        <button
+                                            type="button"
+                                            onClick={(e) => {
+                                                e.stopPropagation()
```

Comment:
Inline functions if possible please replace

Reply:
Won't be able to do it. Passing `mat` value from iteration.